src/credentials/CHIPCert.cpp

```diff
@@ -935,5 +935,47 @@ CHIP_ERROR ConvertECDSASignatureRawToDER(P256ECDSASignatureSpan rawSig, ASN1Writ
     return err;
 }
 
+CHIP_ERROR ExtractPeerIdFromOpCert(const ChipCertificateData & opcert, PeerId * peerId)
```

Comment:
Please add unit tests for this. It should be easy, since we already have test NOCs available.

Reply:
Added tests; good catch.